elastalert/kibana_external_url_formatter.py

```diff
@@ -121,18 +134,29 @@ def create_kibana_auth(kibana_url, rule) -> AuthBase:
     # Unauthenticated
     return None
 
+def is_kibana_atleastsevensixteen(version: str):
+    """
+    Returns True when the Kibana server version >= 7.16
+    """
+    major, minor = list(map(int, version.split(".")[:2]))
+    return major > 7 or (major == 7 and minor >= 16)
 
 def create_kibana_external_url_formatter(
     rule,
     shorten: bool,
-    security_tenant: str
+    security_tenant: str,
 ) -> KibanaExternalUrlFormatter:
     '''Creates a Kibana external url formatter'''
 
     base_url = rule.get('kibana_url')
+    new_shortener = is_kibana_atleastsevensixteen(rule.get('kibana_discover_version', '0.0'))
 
     if shorten:
         auth = create_kibana_auth(base_url, rule)
-        return ShortKibanaExternalUrlFormatter(base_url, auth, security_tenant)
+
+        # id must be unique, can be used to delete the shortened url, currently unused except for creation
+        id = datetime.now().isoformat()
```

Comment:
What about generating a guid instead?  Timestamp might not be precise enough here on a faster computer

Reply:
Actually you can do away with this `id`.  Use the constant `LEGACY_SHORT_URL_LOCATOR` above.